src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String EXEPTION_MESSAGE = "Invalid index";
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            newNode.next = null;
+            newNode.prev = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNode(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        if (index == 0) {
+            first = newNode;
+        } else {
+            isIndexValid(index);
+            currentNode.prev.next = newNode;
+        }
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> currentNode = getNode(index);
+        T oldItem = currentNode.item;
+        currentNode.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexValid(index);
+        Node<T> currentNode = getNode(index);
+        unLink(currentNode);
+        return currentNode.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = first;
+        for (int i = 0; i < size; i++) {
+            if (currentNode.item == object
+                    || (object != null && object.equals(currentNode.item))) {
+                remove(i);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void isIndexValid(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException(EXEPTION_MESSAGE);
+        }
+    }
+
+    private Node<T> getNode(int index) {
```

Comment:
```suggestion
    private Node<T> getNodeByIndex(int index) {
```
Let's rename like this